quote-book/synchronized/src/lib.rs

```diff
@@ -76,6 +76,31 @@ impl<Q: QuoteBook, L: Ledger + Clone + Sync + 'static> SynchronizedQuoteBook<Q,
     pub fn get_current_block_index(&self) -> u64 {
         self.highest_processed_block_index.load(Ordering::SeqCst)
     }
+
+    fn validate_quote_ring_members(&self, quote: &Quote) -> Result<(), Error> {
+        let indices = &quote.sci().tx_out_global_indices;
+        let ring = &quote.sci().tx_in.ring;
+        //We should have an index per txo.
+        if ring.len() != indices.len() {
+            return Err(Error::ImplementationSpecific(
+                "Missing index for ring".to_owned(),
```

Comment:
maybe we can add dedicated error types for this? "InvalidRingMember" or something?